src/ui/common/src/components/pages/workflows/index.tsx

```diff
@@ -148,13 +149,25 @@ const WorkflowsPage: React.FC<Props> = ({ user, Layout = DefaultLayout }) => {
         apiKey: user.apiKey,
         workflowId: workflowId,
       });
+
+      const {
+        data: dags,
+        error: dagsError,
+        isLoading: dagsLoading,
+      } = useDagsGetQuery({
+        apiKey: user.apiKey,
+        workflowId: workflowId,
+      });
```

Comment:
I'd suggest skipping this unless `dagResults` is succeeded but empty, in this way we can have one less API call for workflows already have results